### PR TITLE
Add pretty urls

### PR DESCRIPTION
**Pull request by CamBrown101**
- Installed friendly_Id
- Added slugs to posts
- Added auto incrementing slugs for duplicates
- Fixed broken tests
- Added new tests for routes with slugs

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -40,6 +40,7 @@ group :development, :test do
   gem 'launchy', '~> 2.4.3'
   gem 'rails-controller-testing'
   gem 'faker', :git => 'https://github.com/faker-ruby/faker.git', :branch => 'master'
+  gem 'friendly_id', '~> 5.4.0'
```
Comment (aarondufall) [inline Gemfile:43]: Would you know how to do this with a Gem? Do you how this gem works and what trade-offs you are making?

File: spec/controllers/post_controller_spec.rb [hunk 2]
```diff
@@ -100,14 +100,23 @@
 
     describe "#show" do
       context "with valid attributes" do
-        it "returns a 200" do
+        it "returns a 200 with post id" do
```
Comment (aarondufall) [inline spec/controllers/post_controller_spec.rb:103]: Is calling parameter `id` clear enough? Does it indicate the design choice you have made?

Comment (CamBrown101) [reply spec/controllers/post_controller_spec.rb:103]: Good point. I will fix this and make it more clear.

File: Gemfile (hunk 1 above)
Comment (CamBrown101) [reply Gemfile:43]: I think the trade-offs are not complete control. The positives are they make it pretty easy. I think with google I could do it without the gem. I started to then run into issues with duplicate slugs then found this solution.

**Comment by CamBrown101** [comment]
Closed as I redid this without gem

**Pull request closed without merging**